src/fparser/two/tests/utils/test_sequencebase.py

```diff
@@ -0,0 +1,159 @@
+        assert ("SequenceBase class match method argument string expected to "
+                "be a string but found " in str(excinfo.value))
+
+
+def test_match_invalid_keepempty(f2003_create):
+    '''Test the sequencebase match method raises the expected exception if
+    arg keepempty is not a bool.
+
+    '''
+    for option in [None, 12, Name]:
+        with pytest.raises(InternalError) as excinfo:
+            _ = SequenceBase.match(",", Name, "hello", keep_empty=option)
+        assert ("SequenceBase class match method optional argument keep_empty "
+                "expected to be a boolean but found " in str(excinfo.value))
+
+
+def test_match(f2003_create):
+    '''Test the sequencebase match method with 1 or more entries.'''
+    separator = ","
+    subcls = Name
+    string = "a"
+    result = SequenceBase.match(separator, subcls, string)
+    assert str(result) == "(',', (Name('a'),))"
+    string = "a,b"
+    result = SequenceBase.match(separator, subcls, string)
+    assert str(result) == "(',', (Name('a'), Name('b')))"
+    string = "a,b,c"
+    result = SequenceBase.match(separator, subcls, string)
+    assert str(result) == "(',', (Name('a'), Name('b'), Name('c')))"
+
+
+def test_match_repmap(f2003_create):
+    '''Test the sequencebase match method works if the supplied separator
+    is also used within the matching class.
+
+    '''
+    from fparser.two.Fortran2003 import Entity_Decl_List
+    separator = ","
+    subcls = Entity_Decl_List
+    string = "a(n, n), b(n, n)"
+    result = SequenceBase.match(separator, subcls, string)
+    assert (str(result) ==
+            "(',', (Entity_Decl_List(',', (Entity_Decl(Name('a'), "
+            "Explicit_Shape_Spec_List(',', (Explicit_Shape_Spec(None, "
+            "Name('n')), Explicit_Shape_Spec(None, Name('n')))), None, "
+            "None),)), Entity_Decl_List(',', (Entity_Decl(Name('b'), "
+            "Explicit_Shape_Spec_List(',', (Explicit_Shape_Spec(None, "
+            "Name('n')), Explicit_Shape_Spec(None, Name('n')))), None, "
+            "None),))))")
+
+
+def test_match_keepempty(f2003_create):
+    '''Test the sequencebase match method keep_empty option behaves as
+    expected when set to True or False and when left to its default
+    value. This also tests that a NoMatchError exception occurs if one
+    or more of the entries does not match with the supplied subclass.
+
+    '''
+    separator = " "
+    subcls = Name
+    # Two spaces which means an empty entry between the spaces.
+    string = "a  b"
+
+    # Default should be True so should fail to match.
+    with pytest.raises(NoMatchError):
+        _ = SequenceBase.match(separator, subcls, string)
+
+    # Set to True so should fail to match.
+    with pytest.raises(NoMatchError):
+        _ = SequenceBase.match(separator, subcls, string, keep_empty=True)
+
+    # Set to False so should match.
+    result = SequenceBase.match(separator, subcls, string, keep_empty=False)
```

Comment:
But the empty entry is not 'kept' which is what I was expecting given the name of the `keep_entry` argument. I think this should therefore be renamed to something like `ignore_empty`.

Reply:
Agreed. I've renamed.